### ISSUE TITLE
WooPay custom message sometimes not rendered correctly

### ISSUE THREAD

**Issue by frosso** [Automattic/woocommerce-payments#7302]
### Describe the bug

Reported here https://github.com/Automattic/woocommerce-payments/issues/7196#issuecomment-1736404563

![Screenshot 2023-09-27 at 10 12 51 AM](https://github.com/Automattic/woocommerce-payments/assets/273592/973dbd5d-2915-4952-976a-82abe82b7aef)

It looks like _sometimes_ the custom message for the hosted checkout might not be displayed correctly.
I tried to reproduce this locally with no success. But the site in question does seem to show the incorrect message.
The site in question is p1695762816630409/1695757868.011569-slack-C022WMN88KG

### To Reproduce

Unclear

### Actual behavior

Custom message shows `[terms_of_service_link]` and `[privacy_policy_link]` on hosted checkout
![Screenshot 2023-09-27 at 10 14 52 AM](https://github.com/Automattic/woocommerce-payments/assets/273592/770cf199-21f2-474f-94b2-79b949a22559)

### Screenshots

![Screenshot 2023-09-27 at 10 12 51 AM](https://github.com/Automattic/woocommerce-payments/assets/273592/973dbd5d-2915-4952-976a-82abe82b7aef)

### Expected behavior

Custom message should consistently replace the "shortcode" keys.

### Desktop (please complete the following information):

N/A

### Smartphone (please complete the following information):

N/A

**Comment by pierorocca** [comment]
I do see this on my test site and both my terms and privacy pages are set:

![image](https://github.com/Automattic/woocommerce-payments/assets/109928953/86eca8f8-3568-49d3-b843-960100ecd7f0)

![image](https://github.com/Automattic/woocommerce-payments/assets/109928953/2262ac14-7bf6-4953-b6db-94c097628a0e)

**Comment by pierorocca** [comment]
https://proccaatomic.wpcomstaging.com/

**Comment by pierorocca** [comment]
@frosso resaving my settings resolved the issue if that helps narrow down the issue.

**Comment by gpressutto5** [comment]
@pierorocca I could only reproduce this issue by downgrading WooPayments to a version below 6.4.0 (when this feature was introduced). Do you remember if you were switching versions when this happened?

**Comment by pierorocca** [comment]
@gpressutto5 my test site is Atomic so the version is managed and updated as soon as the new version is available. I made no changes manually. IIRC it only resolved when I specifically saved the settings (without any changes). The two scenarios worth testing are a store on an older WooPayments version and a new build.

**Comment by gpressutto5** [comment]
On Oct 11, you reported this issue on your site. Then, on Oct 12, you resaved the settings, and it was resolved.
I checked the Audit Trail and noticed the WooPayments version of your staging site was upgraded from 6.5.1 to 6.6.0 on Oct 12 before you saved the settings. So, I tested some different versions and confirmed it is an issue in versions 6.5.0 and 6.5.1, but it was fixed in version 6.6.0 (#7225). Resaving the settings was just a coincidence and is not needed.
